students/C_Farris/session05/Mailroom_3.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env Python3
+
+"""
+Date: November 13th, 2018
+Created by: Carol Farris
+Purpose: Mailroom 3
+Goal: Add as many exception handlers as possible (added 3 for User input)
+      Added one comprehension: makereport()
+      update code to continue to keep clean PEP8 code (ongoing)
+Note: I chose to only do one comprehension as I am seeing I will need to refactor my code for testability anyway!
+      I will shoot to add more comprehensions in Mailroom 4
+"""
+
+import sys
+import os
+from textwrap import dedent
+
+OUT_PATH = "thank_you_letters"
+
+
+def prepare_to_write_to_disk():
+    """
+    Check OUT_PATH specified is a directory, if not, it will make it
+    This is to run once at the start of the application.
+    :Param: none
+    :Return: none. Creates specified folder in CWD if not already created.
+    """
+    if not os.path.isdir(OUT_PATH):
+        os.mkdir(OUT_PATH)
+
+
+def send_file_to_disk(person, myfile):
+    """
+    Accepts test file and writes to disk
+    :param (myfile):
+    :return: none. Will send file to directory set in outpath.
+    """
+    filename = person.replace(' ', '_') + '.txt'
+    filename = os.path.join(OUT_PATH, filename)
+    open(filename, 'w').write(myfile)
+
+
+def assemble_thank_you(getPerson='No_name_given',
+                       getDonation='No cash specified'):
+    """
+    assembles a thank you letter
+    :param(getPerson):
+    :param(getDonation):
+    :return: assembled text file with persons full name.txt
+    """
+    return dedent(
+        '''\tDear {},
+        Thank you for your generous donation of ${} to our cause
+        Because of donors like you, we are able to execute our mission
+        to support at risk youth in achieving academic success.
+        Your contribution also ensures we will have the funds to develop
+        new programs to reach those most vulnerable.
+        
+        Sincerely,
+        Team Umizoomi'''.format(getPerson, getDonation))
+
+
+def getDonation():
```

Comment:
All functions should be snake_case - looks like the ones above follow this convention, but then stopped from here down...